components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java

```diff
@@ -30,6 +30,7 @@
 import org.apache.camel.component.kafka.consumer.errorhandler.KafkaConsumerListener;
 import org.apache.camel.health.HealthCheckAware;
 import org.apache.camel.health.HealthCheckHelper;
+import org.apache.camel.impl.health.ComponentsHealthCheckRepository;
```

Comment:
Avoid importing impl classes as they should not be used in the public

Reply:
this is because the ComponentsHealthCheckRepository has an add/remove health check methods: should we create a WritableHealthCheckRepository interface ?

Reply:
Pushed a new version, however the Kafka Consumer/Producer HealthCheck  are also dependant on on an impl `org.apache.camel.impl.health.AbstractHealthCheck`

Reply:
yeah the entire classes in camel-health should be moved to another package - as org.apache.camel.impl is for the core routing engine and camel itself.

Reply:
The `impl` is also in camel-cluster / camel-cloud so its more general. Therefore in light of this then its okay in 3.x to import the impl - We can revisit this in 4.0